geoh5py/objects/cell_object.py

```diff
@@ -89,16 +107,66 @@ def remove_vertices(self, indices: list[int]):
         vert_index[indices] = False
         vertices = self.vertices[vert_index, :]
 
-        # Pre-load data values
-        for child in self.children:
-            if isinstance(child, Data):
-                getattr(child, "values")
-
         self._vertices = None
         setattr(self, "vertices", vertices)
-        self.remove_children_values(indices, "VERTEX")
+        self.remove_children_values(indices, "VERTEX", clear_cache=clear_cache)
 
         new_index = np.ones_like(vert_index, dtype=int)
         new_index[vert_index] = np.arange(self.vertices.shape[0])
         self.remove_cells(np.where(~np.all(vert_index[self.cells], axis=1)))
         setattr(self, "cells", new_index[self.cells])
+
+    def copy(
+        self,
+        parent=None,
+        copy_children: bool = True,
+        clear_cache: bool = False,
+        mask: list[float] | np.ndarray | None = None,
+        cell_mask: list[float] | np.ndarray | None = None,
```

Comment:
**thought (maintainability, performances, for-later):** I suppose `mask` and `cell_mask` always have to be consistent consistent together.
This could be enforced by having `class Mask`, with properties `cell_indices` and `vertex_indices`.

Possibly even subclassing?
- `class VertexMask` with `vertex_indices` only
- `class CellMask` with both `vertex_indices` and `cell_indices`

BTW, also consider using bitarray. The `Mask` class would hide the actual implementation, so that the choice of regular array or bitarray can be revisited later without touching the client code.

`bitarray` will beat `list[int]` in storage efficiency as soon as more than 1/32 of vertices are masked.
On object with n vertices, masking k vertices:
- bitarray takes n bits in memory
- list[int] takes 32*k bits in memory

Masking all or none would take special handling to not store a full bitarray of 0 or 1. Something like
```py
class Mask:
   def __init__(self):
       self.vertex_mask : bitarray | None  = None  # None equivalent to all 0
       self.vertex_mask_all : bool  = false  # if true, equivalent ot bitarray with all 1
```

Reply:
Will implement at the same time as GEOPY-776

Reply:
Will be tackled at the same time as GEOPY-776